core/src/main/java/org/openstack4j/openstack/senlin/domain/SenlinAction.java

```diff
@@ -0,0 +1,109 @@
+package org.openstack4j.openstack.senlin.domain;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.openstack4j.model.senlin.Action;
+import org.openstack4j.openstack.common.ListResult;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * This is a model of a senlinAction. It uses Jackson annotations for
+ * (de)serialization into JSON format
+ * 
+ * @author lion
+ * 
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class SenlinAction implements Action {
+	private static final long serialVersionUID = -1125919503657762374L;
+
+	@JsonProperty("action")
+	private String action;
+	@JsonProperty("cause")
+	private String cause;
+	@JsonProperty("created_at")
+	private String created_at;
+	@JsonProperty("depended_by")
+	private ArrayList depended_by;
+	@JsonProperty("depended_on")
+	private ArrayList depended_on;
```

Comment:
Please use interfaces to represent collection; please refer to existing domain classes